server/src/controllers/authentication.ts

```diff
@@ -0,0 +1,22 @@
+import { Request, Response } from 'express';
+import { compare } from 'bcrypt';
+import { User } from '../models';
+import { loginValidation } from '../validation';
+import { generateToken } from '../helpers';
+
+const loginMiddleware = async (req: Request, res: Response) => {
+  const { email, password } = req.body;
+  await loginValidation.validate({ email, password });
+
+  const user = await User.findOne({ where: { email } });
+  if (!user) return { status: 400, msg: 'User not found' };
```

Comment:
```suggestion
  if (!user) throw serverErrs.BadReq
```
Also, I'd prefer to send hazy message, that email or password incorrect
Read more on [Authentication and error messages](https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html#authentication-and-error-messages)